### PR TITLE
CASSANDRA-18987 Prohibit counter column updates in Accord transactions

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe; reviewed by David Capwell for CASSANDRA-18987

### CODE REVIEW

File: src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java [hunk 1]
```diff
@@ -84,9 +84,7 @@
 
 import static accord.primitives.Txn.Kind.EphemeralRead;
 import static accord.primitives.Txn.Kind.Read;
-import static org.apache.cassandra.cql3.statements.RequestValidations.checkFalse;
-import static org.apache.cassandra.cql3.statements.RequestValidations.checkNotNull;
-import static org.apache.cassandra.cql3.statements.RequestValidations.checkTrue;
+import static org.apache.cassandra.cql3.statements.RequestValidations.*;
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java]: can you revert?  We don't like `*` in our style guide

File: src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java [hunk 2]
```diff
@@ -556,6 +555,10 @@ public CQLStatement prepare(ClientState state)
                 checkFalse(prepared.hasConditions(), NO_CONDITIONS_IN_UPDATES_MESSAGE, prepared.type, prepared.source);
                 checkFalse(prepared.isTimestampSet(), NO_TIMESTAMPS_IN_UPDATES_MESSAGE, prepared.type, prepared.source);
 
+                for (ColumnMetadata column : prepared.updatedColumns())
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java]: what if a reference is to a counter type?

```
LET a = (select counter from tbl where pk=?);
IF a.counter > 10 THEN
  ...
END IF
```

File: test/unit/org/apache/cassandra/cql3/statements/TransactionStatementTest.java [hunk 3]
```diff
@@ -33,13 +33,7 @@
...
 import org.assertj.core.api.Assertions;
 
-import static org.apache.cassandra.cql3.statements.TransactionStatement.DUPLICATE_TUPLE_NAME_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.EMPTY_TRANSACTION_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.ILLEGAL_RANGE_QUERY_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.INCOMPLETE_PRIMARY_KEY_SELECT_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.NO_CONDITIONS_IN_UPDATES_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.NO_TIMESTAMPS_IN_UPDATES_MESSAGE;
-import static org.apache.cassandra.cql3.statements.TransactionStatement.SELECT_REFS_NEED_COLUMN_MESSAGE;
+import static org.apache.cassandra.cql3.statements.TransactionStatement.*;
```
Comment (dcapwell) [inline test/unit/org/apache/cassandra/cql3/statements/TransactionStatementTest.java]: can you revert?  `*` isn't desired in our style guides

**Review by dcapwell — CHANGES_REQUESTED** [review]

**Review by dcapwell — APPROVED** [review]

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/c876f6a01dd9ac254406898cef0727694b38d45a

**Pull request closed without merging**